### PR TITLE
feat(config): add safeguard timeouts 

### LINKED ISSUE #2804: Add safeguard timeouts
Add a safeguard timeout for each child process we call, including Docker. Ideally configured as an object that includes type: value mappings, e.g. `"npm": 600` meaning 600s timeout for npm install.

When using Docker, it should ideally kill the container too, if the timeout occurs.

### PR DESCRIPTION
**Pull request by olegkrivtsov**
## Changes:

Add safeguard timeouts 
Closes #2804
Replaces #12274

## Context:

- Closes #2804
- Closes #12274

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/util/exec/index.ts [hunk 1]
```diff
@@ -79,7 +80,8 @@ function getRawExecOptions(opts: ExecOptions): RawExecOptions {
     cwd,
   };
   // Set default timeout to 15 minutes
-  rawExecOptions.timeout = rawExecOptions.timeout || 15 * 60 * 1000;
+  rawExecOptions.timeout =
+    rawExecOptions.timeout || defaultExecutionTimeout || 15 * 60 * 1000;
```
Comment (viceice) [inline lib/util/exec/index.ts:84]:
```suggestion
    rawExecOptions.timeout ?? defaultExecutionTimeout ?? 15 * 60 * 1000;
```

**Review by viceice — COMMENTED** [review]
Looks good but don't fully meets the requirements of the issue to be closed.

**Review by HonkingGoose — COMMENTED** [review]
Docs look good to me.

File: lib/util/exec/index.ts [hunk 2]
```diff
@@ -79,7 +80,8 @@ function getRawExecOptions(opts: ExecOptions): RawExecOptions {
     cwd,
   };
   // Set default timeout to 15 minutes
-  rawExecOptions.timeout = rawExecOptions.timeout || 15 * 60 * 1000;
+  rawExecOptions.timeout =
+    rawExecOptions.timeout ?? defaultExecutionTimeout ?? 15 * 60 * 1000;
```
Comment (rarkins) [inline lib/util/exec/index.ts:84]: Isn't this defaultExecutionTimeout in minutes, not milliseconds?

**Comment by rarkins** [comment]
Seems like this and https://github.com/renovatebot/renovate/pull/12274 are duplicated

File: lib/util/exec/index.ts (hunk 2 above)
Comment (olegkrivtsov) [reply lib/util/exec/index.ts:84]: It sounds you are right, I'll fix this

**Review by viceice — DISMISSED** [review]

**Review by viceice — DISMISSED** [review]

**Review by viceice — APPROVED** [review]

**Pull request merged**